styx-api-service/src/main/java/com/spotify/styx/api/Middlewares.java

```diff
@@ -67,4 +73,21 @@ private Middlewares() {
       }
     });
   }
+
+  public static Middleware<AsyncHandler<? extends Response<?>>,
+      AsyncHandler<? extends Response<ByteString>>> clientValidator(
+      Supplier<Optional<List<String>>> supplier) {
+    return innerHandler -> requestContext -> {
```

Comment:
We don't have any metric emitted from styx-api-service. This feels like a separated PR. I will add todo for now.